website/src/AppFrame.js

```diff
@@ -36,26 +43,105 @@ const NavigationLinks = ({navigation, className, reverse}) => {
   );
 }
 
+class VersionSelector extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { menuShown: false };
+
+    this.toggleMenuShown = this.toggleMenuShown.bind(this);
+    this.handleVersionClick = this.handleVersionClick.bind(this);
+  }
+
+  toggleMenuShown() {
+    this.setState({ menuShown: !this.state.menuShown });
+  }
+
+  handleVersionClick(version) {
+    const { setParams } = this.props.childNavigation;
+    return () => {
+      setParams({ version });
+    }
+  }
+
+  render() {
+    const { menuShown } = this.state;
+    const { state } = this.props.childNavigation;
+    
+    const isUndef = (obj) => obj === undefined;
+    const docVersionIsUndefined = (state) => (
+      isUndef(state) || 
+      isUndef(state.params) || 
+      isUndef(state.params.version)
+    );
+
+    const currentVersion = docVersionIsUndefined(state) ? 'edge' : state.params.version;
+    const versionsList = Object.keys(docs).map((version) => (
+      <li
+        key={version}
+        onClick={this.handleVersionClick(version)}
+        style={{
+          color: (currentVersion === version) ? '#6b52ae' : '#555'
+        }}
+      >
+        {version}
+      </li>
+    ));
+
+    return (
+      <div className='versionselector'>
+        <span className="versionbutton" onClick={this.toggleMenuShown}>
+          version: {currentVersion} ▼
+        </span>
+        <ul
+          style={{
+            //display: menuShown ? 'inline' : 'none',
+            visibility: menuShown ? 'visible' : 'hidden',
+            listStyleType: 'none',
+          }}
+        >
+          {versionsList}
+        </ul>
+      </div>
+    );
+  }
+}
+
 class AppFrame extends React.Component {
-  state = {isMobileMenuOpen: false};
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isMobileMenuOpen: false,
+    };
+
+    this.setVersion = this.setVersion.bind(this);
+  }
+  
   componentWillReceiveProps(props) {
     if (this.props.navigation.state !== props.navigation.state) {
       this.setState({isMobileMenuOpen: false});
       window.scrollTo(0,0);
     }
   }
+  
+  setVersion(version) {
+    return () => this.setState({ version });
```

Comment:
Same as previously, no need to return function here. Prefer `onPress={() => setVersion(version)}` instead